lib/SILOptimizer/Mandatory/DiagnoseStaticExclusivity.cpp

```diff
@@ -718,20 +734,20 @@ static void checkForViolationWithCall(
     if (SubAccesses.empty())
       continue;
 
-    SILValue Argument = Arguments[ArgumentIndex];
+    SILValue Argument = Apply.getArgument(ArgumentIndex);
     assert(Argument->getType().isAddress());
 
     const AccessedStorage &Storage = findValidAccessedStorage(Argument);
-    auto AccessIt = Accesses.find(Storage);
+    auto AccessIt = State.Accesses->find(Storage);
 
     // Are there any accesses in progress at the time of the call?
-    if (AccessIt == Accesses.end())
+    if (AccessIt == State.Accesses->end())
       continue;
 
     const AccessInfo &Info = AccessIt->getSecond();
     if (auto Conflict = findConflictingArgumentAccess(AS, Storage, Info)) {
-      Conflict->setAlwaysDiagnoseAsWarning(DiagnoseAsWarning);
-      ConflictingAccesses.push_back(*Conflict);
+      Conflict->setAlwaysDiagnoseAsWarning(/*DiagnoseAsWarning=*/false);
```

Comment:
I agree with you that it should never be a warning, but, why keep the callee function as one taking a bool? Looking at the code base this is the only call site - can simplify the callee as well.

Reply:
Fixed.